Updating README.md with a couple missing flags. (#198)
cyberark/summon: README.md

    summon -D ENV=production --yaml 'SQL_PASSWORD: !var env/$ENV/db-password' deploy.sh
    ```

* `--yaml` secrets.yml as a literal string.

>     This flag is used to put the secrets.yml variables on the command line (see example above).

Reviewer: Codeclimate is complaining about line length here

Author: updated

>     This flag is used to put the secrets.yml variables on the command line (see example above).

Reviewer: ```suggestion
    This flag is used to pass `secrets.yml` to the provider as a literal string (see example above).
```

Author: updated with suggestions.

    summon -D ENV=production --yaml 'SQL_PASSWORD: !var env/$ENV/db-password' deploy.sh
    ```

> * `--yaml` secrets.yml as a literal string.

Reviewer 2: It might be clearer to add syntax to point out the expected literal YAML. Also, might be clearer to say that the literal YAML replaces `secrets.yml` file (it could be interpreted sorta reads as if the `secrets.yml` file could be getting passed here), something like:
```
`--yaml <YAML-string>` Passes secrets.yml as a literal string

Passes a literal YAML string to be parsed by the provider in place of a
`secrets.yml` file (see example above).
```

Also, Line 151 (not in your change set) could also similarly use a `<path-to-provider>` field (https://github.com/cyberark/summon/pull/198/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R151).

Same for Line 182 for `-i, --ignore` (https://github.com/cyberark/summon/pull/198/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R182)

Author: ok, Will add these to a follow-on commit.

    This flag can be useful when the underlying system that's going to be using the values implements defaults. For example, when using summon as a bridge to [confd](https://github.com/kelseyhightower/confd).

> * `--all-provider-versions, -V` List of all of the providers in the default path and their versions

Reviewer: Codeclimate is complaining about line length

Author: updated.

> * `--all-provider-versions, -V` List of all of the providers in the default path and their versions

Reviewer: ```suggestion
* `-V, --all-provider-versions` List of all of the providers in the default path and their versions
```

It looks like the formatting we use typically has the abbreviated flag listed first

Author: ok

    (if they have the --version tag).

> * `--version, -v` Print the version.

Reviewer: ```suggestion
* `-v, --version` Print the Summon version.
```

Author: got this one too
